libopenage/util/compiler.h

```diff
@@ -8,18 +8,44 @@
  *
  * May contain some platform-dependent code.
  */
+ #include <ciso646>
 
 // pxd: from libcpp.string cimport string
 #include <string>
 
+/**
+ * DLL entry-point decorations.
+ */
+#if defined(_WIN32)
+	#if defined(libopenage_EXPORTS)
+		#define LIBOPENAGE_API __declspec(dllexport)
+	#else
+		#define LIBOPENAGE_API __declspec(dllimport)
+	#endif /* libopenage_EXPORTS */
+#else
+	#define LIBOPENAGE_API
+#endif
```

Comment:
AFAIK, this is the only way to sanely* link to a DLL('s import library) with C++ interfaces in VS.
The alternatives while still using a shared library include:
- Restricting to C-style interfaces (like extern "C" foo). Then using `LoadLibrary` and `GetProcAddress` (equivalents of `dlopen` and `dlsym` respectively) for using the DLL at runtime.
- Go completely off-book and implement mangling for C++ identifiers (which is bound to break in the next release of VS).

Another alternative is to change libopenage to a static library - the problem would be it'll be linked to every python extension (.pyd) we build - bloating up the package when we build one.

Usually this macro doesn't scatter as much if the interface is small and consolidated. In our case, the interface with python is spread across the library.

*Sanity is relative. Essentially an alien concept to VS users.